src/cultionet/callbacks.py

```diff
@@ -0,0 +1,163 @@
+import typing as T
+import filelock
+from pathlib import Path
+
+import geowombat as gw
+import rasterio as rio
+from rasterio.windows import Window
+import torch
+from pytorch_lightning.callbacks import BasePredictionWriter
+from torch_geometric.data import Data
+
+from .data.const import SCALE_FACTOR
+from .utils.reshape import ModelOutputs
+
+
+class LightningGTiffWriter(BasePredictionWriter):
```

Comment:
This is a `pytorch_lightning` callback that is used to predict and write predictions to a raster file.